x/validator-preference/keeper/ValidatorSet.go

```diff
@@ -0,0 +1,45 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	"github.com/osmosis-labs/osmosis/v12/x/validator-preference/types"
+)
+
+// ValidateValidator checks if the validator address is valid and the validator provided exists onchain.
+func (k Keeper) ValidateValidator(ctx sdk.Context, valOperAddress string) (sdk.ValAddress, stakingtypes.Validator, error) {
```

Comment:
nit: I feel like this should be renamed to `GetValAddrAndVal` or something similar, since a method starting with Validate gives us the impression of either returning a bool, or that it's purpose is simply validating and returning errors. Might be my personal view tho, feel free to skip if u have different opinions

Reply:
Gotchu, this method validates and returns (valAddr, Val) so i think renaming to `GetValAddrAndVal` makes sense